Fix forced powering off

nest1ing opened the pull request.

nest1ing committed:
Fix misspelling

This fixes a misspelling in the host power off option.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

artemsen commented on commands/host.vegman:
```diff
@@ -98,12 +98,11 @@ function cmd_power_off {
     ipmitool power soft
-    echo "Shutdown signal sent to host, waiting to power the system off."
   fi
+  echo "Shutdown signal sent to host, waiting to power the system off."
```
`ipmitool power off` will turn the host off an any case. Why "waiting"?

AlexanderAmelkin replied:
Yup. Besides, nothing in the code is actually 'waiting'.

nest1ing committed:
Fix 2-level subcommands execution

The commit b1dd0544ecceb6c82b6af148f912ef5ae1807873 made the 2-level
subcommands unexecutable.
For example:
```
$ host power status
Invalid argument: power status
```

This commit fixes the issue.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

nest1ing committed:
vegman: bmc datetime set: fix execution

The commit d35d657ca8d5d783d9f5012b8a100c6ed0c7f36a brought a mistype
that makes `bmc datetime set` command already fails.

This commit fixes it.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

nest1ing committed:
vegman: bmc power: use hostpwrctl

This commit makes the `host power` command to use `hostpwrctl` that
waits till the requested state is reached.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

AlexanderAmelkin commented on commands/host.vegman:
```diff
@@ -65,8 +65,7 @@ function cmd_power {
   expect_noarg "$@"
-  ipmitool power on
-  echo "Chassis power is switched on, host is booting up."
+  hostpwrctl on
```
Need `exec -a` for `hostpwrctl`'s help to work properly.
The `hostpwrctl` command itself shouldn't be available to the users, so they need to get help in the form of a CLI command.

AlexanderAmelkin committed:
clicmd: Add a generic exec_tool function

Now, as hostpwrctl is added as another external tool
that may display its own help, remove `call_netconfig`
and add a generic `exec_tool` function instead. Use it
both for `netconfig` and for `hostpwrctl` to substitute
their names by the name of the called CLI command.

Also use it for `fwupdate` for uniformity, although that
tool is called in CLI in a way that should never result
in printing of usage help by the tool itself.

Signed-off-by: Alexander Amelkin <a.amelkin@yadro.com>

nest1ing reviewed (commented):
Looks good to me, but github doesn't allow me to approve it as I'm an author of the PR.

nest1ing commented:
And this PR requires https://github.com/YADRO-KNS/obmc-yadro-hostpwrctl/pull/1
